### ISSUE TITLE
Nested form : Can't reach a nested entity

### ISSUE THREAD

**Issue by PxlCtzn** [sonata-project/SonataAdminBundle#2716]
Hi folks,

Here is my pb: I can not reach the variable 'label' of the EquipmentML entity inside the *Edit hostel* form.
**tl;dr** : An hostel contains multiple Equipments (oneToMany) (joinTable : HostelEquipment). Each HostelEquipment can be visible or not (that's why I set up the joinTable manually). Moreover the name of the Equipment is available in various language (5 of them). To do so, the Equipment are linked to the EquipmentML entity.
So basically : Hostel  -> HostelEquipment -> Equipment -> EquipmentML
I'm stuck to the Equipment. I can get all the equipments but not the EquipmentML associated.

I rtfm but I did not find anything on this specific topic so all the help will be appreciate.

I have the following entities : 
**Hostel** (tl;dr: *describes an hostel*)
```php
/**
 * Hostel
 * @ORM\Entity(repositoryClass="AppBundle\Repository\HostelRepository")
 * @ORM\Table(name="hostel")
 * UniqueEntity("code")
 */
class Hostel
{
    /**
     * @var integer
     *
     * @ORM\Column(name="id", type="integer")
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="AUTO")
     */
    private $id;
    // [...]
    /**
     * @var ArrayCollection
     * @ORM\OneToMany(targetEntity="HostelEquipment", mappedBy="hostel" , cascade={"persist", "remove"})
     * @ORM\JoinColumn(name="equipment_id", referencedColumnName="id")
     */
    private $hostelEquipments;
}
```
**HostelEquipment** (tl;dr: *link an equipment to an hostel + contains a boolean to set if the equipment is visible or not*)
```php
/**
 * Describe an Equipment
 * @ORM\Entity
 * @ORM\Table(name="hostel_equipment")
 */
class HostelEquipment
{
    /**
     * ManyToOne Unidirectional
     * @ORM\Id
     * @ORM\ManyToOne(targetEntity="Hostel", inversedBy="hostelEquipments", cascade={"persist", "remove"})
     * @ORM\JoinColumn(name="hostel_id", referencedColumnName="id")
     */
    private $hostel;

    /**
     * @ORM\Id
     * @var Equipment $equipment
     * ManyToOne Unidirectional
     * @ORM\ManyToOne(targetEntity="Equipment",  cascade={"persist", "remove"})
     * @ORM\JoinColumn(name="equipment_id", referencedColumnName="id")
     */
    private $equipment;

    /**
     * @var boolean
     * @ORM\Column(name="status", type="boolean");
     */
    private $status;
    // [...]
}
```
**Equipment** (tl;dr: *one equipment*)
```php
/**
 * Describe an Equipment
 * @ORM\Entity
 * @ORM\Table(name="equipment")
 */
class Equipment
{
    /**
     * @var integer
     *
     * @ORM\Column(name="id", type="integer")
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="AUTO")
     */
    private $id;

    /**
     * @var ArrayCollection $equipmentML
     * @ORM\OneToMany(targetEntity="EquipmentML", mappedBy="equipment")
     * @ORM\JoinColumn(name="equipment_multilanguage_id", referencedColumnName="id")
     */
    private $equipmentML;
    // [...]
}
```
**EquipmentML** (tl;dr: *Contains the entity's data for multi-language support*)
```php
/**
 * @ORM\Entity
 * @ORM\Table(name="equipment_multilanguage")
 */
class EquipmentML {
    /**
     * @var LanguageFO $languageFO
     * @ORM\Id
     * @ORM\ManyToOne(targetEntity="LanguageFO")
     * @ORM\JoinColumn(name="language_fo_id", referencedColumnName="id")
     */
    private $languageFO;

    /**
     * @var Equipment $equipment
     * @ORM\Id
     * @ORM\ManyToOne(targetEntity="Equipment", inversedBy="equipmentML")
     * @ORM\JoinColumn(name="equipment_id", referencedColumnName="id")
     */
    private $equipment;

    /**
     * @var string $label
     * @ORM\Column(name="label", type="text")
     */
    private $label;
    // [...]
}
```

Here is my *services.yml*  : 
```yaml
services:
    sonata.admin.hostel:
          class: AppBundle\Admin\HostelAdmin
          tags:
              - { name: sonata.admin, manager_type: orm, group: "hostel", label: "Hostel" }
          arguments:
              - ~
              - AppBundle\Entity\Hostel
              - AppBundle:HostelAdmin
          calls:
              - [ setTranslationDomain,   [AppBundle]]
    sonata.admin.equipment:
          class: AppBundle\Admin\EquipmentAdmin
          tags:
              - { name: sonata.admin, manager_type: orm, group: "hostel", label: "Equipment" }
          arguments:
              - ~
              - AppBundle\Entity\Equipment
              - ~
          calls:
              - [ setTranslationDomain,   [AppBundle]]
    sonata.admin.equipmentML:
          class: AppBundle\Admin\EquipmentMLAdmin
          tags:
              - { name: sonata.admin, manager_type: orm, group: "hostel", label: "EquipmentML" }
          arguments:
              - ~
              - AppBundle\Entity\EquipmentML
              - ~
          calls:
              - [ setTranslationDomain,   [AppBundle]]
    sonata.admin.hostelEquipment:
          class: AppBundle\Admin\HostelEquipmentAdmin
          tags:
              - { name: sonata.admin, manager_type: orm, group: "hostel", label: "HostelEquipment" }
          arguments:
              - ~
              - AppBundle\Entity\HostelEquipment
              - ~
          calls:
              - [ setTranslationDomain,   [AppBundle]]
```

And, at last, the Admins classes : 
**HostelAdmin** (tl;dr *Add the 'hostelEquipments' as a sonata_type_collection*)
```php
class HostelAdmin extends Admin {
    protected function configureFormFields(FormMapper $form)    {
        $form->add('hostelEquipments', 'sonata_type_collection');
    }
}
```
**HostelEquipmentAdmin** (tl;dr *Add the 'equipment' as a 'sonata_type_collection'*)
```php
class HostelEquipmentAdmin extends Admin
{
    protected function configureFormFields(FormMapper $form) {
        $form->add('equipment', 'sonata_type_model');
    }
}
```
**EquipmentAdmin** (tl;dr *Add the 'equipmentML' as a 'sonata_type_collection'*)
```php
class EquipmentAdmin extends Admin
{
    protected function configureFormFields(FormMapper $form) {
        $form->add('equipmentML', 'sonata_type_collection');
    }
}
```
**EquipmentMLAdmin** (tl;dr *Add the 'equipmentML' as a 'sonata_type_collection'*)
```php
class EquipmentMLAdmin extends Admin
{
    protected function configureFormFields(FormMapper $form)
    {
        $form->add('label');
    }
}
```

**Comment by dypa** [comment]
refs #1971
